src/caselawclient/models/identifiers/__init__.py

```diff
@@ -127,6 +127,14 @@ def __delitem__(self, key: Union[Identifier, str]) -> None:
         else:
             super().__delitem__(key)
 
+    def delete_type(self, deleted_identifier_type: type[Identifier]) -> None:
```

Comment:
I'm happy this existing for now (because we're being a bit blunt-instrument about NCNs), but when we handle B-citations this should probably go away and be replaced by something giving deletions more precision and intent.